File: Python/Grace/Grace.py
#!/usr/bin/python3
import os
NL=10
VALUE="#!/usr/bin/python3{nl}import os{nl}NL=10{nl}VALUE={quo}{val}{quo}{nl}def main(): out=os.open({quo}Grace_kid.py{quo}, os.O_WRONLY | os.O_TRUNC | os.O_CREAT, 0o744); os.write(out, VALUE.format(nl=chr(NL),quo=chr(34),val=VALUE).encode('UTF-8')){nl}if __name__ == {quo}__main__{quo}:{nl}    ## This is a comment{nl}    try:main(){nl}    except:pass{nl}"
def main(): out=os.open("Grace_kid.py", os.O_WRONLY | os.O_TRUNC | os.O_CREAT, 0o744); os.write(out, VALUE.format(nl=chr(NL),quo=chr(34),val=VALUE).encode('UTF-8'))
if __name__ == "__main__":
    ## This is a comment
    try:main()
    except:pass

File: C/Grace/Grace.c
#include <stdio.h>
#include <unistd.h>
#include <fcntl.h>
#define nl 10
#define val "#include <stdio.h>%3$c#include <unistd.h>%3$c#include <fcntl.h>%3$c#define nl 10%3$c#define val %1$c%2$s%1$c%3$c#define MAIN(x) int main(){int fd=open(%1$cGrace_kid.c%1$c,O_WRONLY|O_TRUNC|O_CREAT,0644);if(fd>-1){dprintf(fd,x,34,x,nl);}close(fd);}%3$c/*%3$c  This is a comment%3$c*/%3$cMAIN(val)%3$c"
#define MAIN(x) int main(){int fd=open("Grace_kid.c",O_WRONLY|O_TRUNC|O_CREAT,0644);if(fd>-1){dprintf(fd,x,34,x,nl);}close(fd);}
/*
  This is a comment
*/
MAIN(val)
